From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: ctcmpc: use correct idal word list for ctcmpc
Patch-mainline: Yes
References: bnc#750173,LTC#79264

Symptom:     traffic hang with Communications Server Linux
Problem:     Sending problems have shown up caused by a wrong idal
             word list for the first ccw. Function
             ctcmpc_chx_txdone() invokes the function to prepare
             the idal word list without setting an appropriate length
             for the first ccw, which may lead to an incomplete idal
             word list.
Solution:    This patch sets the maximum buffer size as data length
             of the first ccw. Thus correct idal word lists are
             guaranteed in all cases.

Acked-by: John Jolly <jjolly@suse.de>

---
 drivers/s390/net/ctcm_fsms.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/s390/net/ctcm_fsms.c
+++ b/drivers/s390/net/ctcm_fsms.c
@@ -1341,6 +1341,7 @@ static void ctcmpc_chx_txdone(fsm_instan
 
 	spin_unlock(&ch->collect_lock);
 	clear_normalized_cda(&ch->ccw[1]);
+	ch->ccw[1].count = ch->max_bufsize;
 	if (set_normalized_cda(&ch->ccw[1], ch->trans_skb->data)) {
 		dev_kfree_skb_any(ch->trans_skb);
 		ch->trans_skb = NULL;
